archivist/runner.py

```diff
@@ -82,113 +109,101 @@ def run_steps(self, config: Dict):
         for step in config["steps"]:
             self.run_step(step)
 
-        self.delete()
-
-    def run_step(self, step: Dict):
+    def run_step(
+        self, step: Dict
+    ):  # pylint:disable=too-many-branches  # This should be fixed
```

Comment:
How urgently should this be fixed? If this comment is likely to be long lived can we have a brief description of why we currently believe we need to do this, and where the cause is?

Reply:
Yeah - the logic in the runner.run_step() method consists of too many if statements to accommodate a wide range of possiblilities - usually this style of code is fragile and requires complicated test scenarios.. i.e the maintenance burden becomes problematic.

Having said that it only just exceeded the limits set in pylintrc configuration file so I am not worried.

I will add a suitable comment

Reply:
this has now been refactored